### Verified end-to-end against the examples repo

Ran `./gradlew publishLocal` on this branch (produced `~/.m2/repository/org/xtclang/xdk/0.4.4-SNAPSHOT/xdk-0.4.4-SNAPSHOT.zip` SHA `c0e0c263...`), enabled `mavenLocal` in the `xtclang/examples` repo's `settings.gradle.kts` + `build-logic/settings.gradle.kts` + `chess-game/settings.gradle.kts` + `webapp-conventions.gradle.kts` so the chess-game build picks up the fixed XDK, then temporarily removed one `@Disabled` from `chess-game/app/src/test/x/ChessPieceValidatorTest.x` (the intentionally-broken `shouldValidateBasicPieceMovement`) and ran:

```
./gradlew :chess-game:app:testXtc --refresh-dependencies
```

**Result with the fix applied:**

```
Finished: AIPositionEvaluatorTests Failed in 0.072 (passed=4 failed=1 skipped=0 errors=0)
Finished: ChessAIEvaluatorTest Failed in 0.072 (passed=4 failed=1 skipped=0 errors=0)
> Task :chess-game:app:testXtc FAILED
  > [plugin] Test failure.
BUILD FAILED in 24s
```

Before the fix (remote `0.4.4-SNAPSHOT` XDK), the exact same code / same failing test produced `BUILD SUCCESSFUL`. After the fix, the xunit engine returns a real exit code, `XtcRunTask.runSingleModule` throws on `exitCode != 0`, `XtcTestTask` re-throws because `failOnTestFailure = true`, and Gradle fails the build. Propagation works.

### Bonus finding

The fix surfaced a previously-hidden failing test that was silently ignored by the old `always-return-True` path:

```
Finished: AIPositionEvaluatorTests.shouldEvaluatePositionWithMaterialAdvantage() Failed in 0.009
IllegalState: "eval > 0": eval=-975
```

Unrelated to this PR — it's a real chess-evaluation bug in `AIPositionEvaluator.x` that was masked by the xunit bug. I'll track it separately in the `examples` repo; mentioning it here only as evidence that the runner is now actually reporting what the tests say.

--- a/plugin/src/main/java/org/xtclang/plugin/XtcProjectDelegate.java
+++ b/plugin/src/main/java/org/xtclang/plugin/XtcProjectDelegate.java
@@ -537,7 +537,12 @@ public class XtcProjectDelegate {
         testTask.configure(task -> {
             task.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
             task.setDescription("Run XTC xunit tests. Skip with -P" + PROPERTY_SKIP_TESTS + " or -P" + PROPERTY_SKIP_ALL_TESTS + ".");
-            task.getFailOnTestFailure().convention(true);
+            // NOTE: do not set task.getFailOnTestFailure().convention(...) here.
+            // The task's constructor already wires its convention to the
+            // xtcTest extension's failOnTestFailure property; setting another
+            // convention at registration time replaces that wiring, which
+            // breaks the `xtcTest { failOnTestFailure = ... }` DSL. The
+            // default value comes from DefaultXtcTestExtension (true).
             task.dependsOn(compileTaskNames);
             task.onlyIf(t -> !shouldSkip);
             if (shouldSkip) {

